src/org.hdfgroup.hdfview/hdf/view/TableView/DataDisplayConverterFactory.java

```diff
@@ -758,10 +760,18 @@ else if (showAsBin) {
                         buffer.append(Tools.toBinaryString(Long.valueOf(value.toString()), (int)typeSize));
                 }
                 else if (numberFormat != null) {
-                    buffer.append(numberFormat.format(value));
+                    log.trace("canonicalToDisplayValue({}): numberFormat", value);
+                    if (isFLT16)
+                        buffer.append(numberFormat.format((float)value));
```

Comment:
Shouldn't these values already have been converted to float by the DataProviderFactory? I'd generally prefer to avoid as many `isFLT16` if statements as possible, especially if we support other non-standard types in the future.

Reply:
Yes, now they are. So might not need one. Will test it.

Reply:
done.